integration_tests/src/main/python/data_gen.py

```diff
@@ -192,19 +194,24 @@ def with_special_pattern(self, pattern, flags=0, charset=sre_yield.CHARSET, weig
         instead of a hard coded string value.
         """
         strs = sre_yield.AllStrings(pattern, flags=flags, charset=charset, max_count=_MAX_CHOICES)
-        try:
-            length = int(len(strs))
-        except OverflowError:
-            length = _MAX_CHOICES
-        return self.with_special_case(lambda rand : strs[rand.randrange(0, length)], weight=weight)
+        length = strs.__len__()
+        def gen_str(rand):
+            index = rand.random() * length
+            return strs[int(index)]
```

Comment:
Okay so the overflow problem was happening because if `len`, but `__len__` bypasses that. I didn't expect that.  At least in my testing with python 3.7.3 `randrange` supports a very large number for length. At least as large as sre_yeild can support

```
>>> import sre_yield
>>> strs = sre_yield.AllStrings(".{0,30}")
>>> strs.__len__()
1577291187349979301006054836464370837822167673657674866721151381965697281
>>> random.randrange(0, strs.__len__())
631767151906816103137124906095165560283366027183596615061544231764767003
>>> random.randrange(0, strs.__len__())
1106341783813440001454877005763897253653655071394856967045913002089358699
>>> random.randrange(0, strs.__len__())
1481356946441039628270334070175674863332509159999467256291081692197656622
>>> strs[random.randrange(0, strs.__len__())]
'Ö\xa0\x10F6\x0e-¾\x952\x92÷·"SÎTw\x14âr\x97ÆÄcgÿ7ê_'
>>> strs[random.randrange(0, strs.__len__())]
'ù÷§®°î\x8fPE8xQ(W\x89è5/\x95\t\x02Ðø¡\x01¤aýÛã'
>>> strs[random.randrange(0, strs.__len__())]
's3N®?KlîNÌÈ)68U\x01\x9c~k\x084æ¬\x1dqZß\x10hn'
```

Perhaps that is the better solution?  Because then we would not limit ourselves to the 56 bits of randomness that rand.random() supports.

Reply:
Thanks, done.